tests/e2e/conftest.py

```diff
@@ -90,3 +103,119 @@ def pytest_addoption(parser):
+    Returns:
+        temporary file's name
+    """
+    with tempfile.NamedTemporaryFile(mode="w", delete=False) as temp_file:
+        json.dump(json_data, temp_file)
+        temp_file.flush()
+        return temp_file.name
+
+
+def create_datarouter_config_file(session):
+    """Create datarouter config file."""
+    project = session.config.getini("rp_project")
+    endpoint = session.config.getini("rp_endpoint").replace("https://", "")
+    launch = session.config.option.rp_name
+    launch_desc = session.config.getini("rp_launch_description") or ""
+    json_data = {
+        "targets": {
+            "reportportal": {
+                "config": {
+                    "hostname": endpoint,
+                    "project": project,
+                },
+                "processing": {
+                    "apply_tfa": True,
+                    "property_filter": ["^(?!(polarion|iqe_blocker).*$).*"],
+                    "launch": {
+                        "name": launch,
+                        "description": launch_desc,
+                    },
+                },
+            }
+        }
+    }
+
+    temp_filename = write_json_to_temp_file(json_data)
+    return temp_filename
+
+
+def add_secret_to_env(env) -> None:
+    """Add aws secrets from environment variable to dict.
+
+    Args:
+        env (env): environment variable name
+    Returns:
+        Null
+    """
+    name = env[:-5]
+    with open(os.environ[env]) as file:
+        content = file.read()
+        aws_env[name] = content
+    return
+
+
+def get_secret_value(env: str) -> str:
+    """Handle secrets delivered in env variables."""
+    with open(os.environ[env]) as file:
+        return file.read()
+
+
+def pytest_sessionfinish(session):
+    """Create datarouter compatible archive to upload into report portal."""
+    # Sending reports to report portal
+    try:
+        datarouter_config = create_datarouter_config_file(session)
+        archive_path = os.path.join(os.getcwd(), f"reportportal-{uuid.uuid4()}.tar.gz")
+        with tarfile.open(archive_path, "w:gz") as tar:
+            tar.add(datarouter_config, arcname="data_router.json")
+            file_path = os.environ["ARTIFACT_DIR"]
+            for file in os.listdir(file_path):
+                if file.endswith(".xml"):
+                    print(f"Found xml to add in archive {file}.")
+                    tar.add(
+                        os.path.join(file_path, file),
+                        arcname=os.path.join("data", "results", file),
+                    )
+        print(f"Saved Report Portal datarouter archive to {archive_path}.")
+    except Exception as e:
+        print(f"Error creating RP archive: {e}")
+        return None
+    try:
+        add_secret_to_env("AWS_ACCESS_KEY_ID_PATH")
+        add_secret_to_env("AWS_BUCKET_PATH")
+        add_secret_to_env("AWS_REGION_PATH")
+        add_secret_to_env("AWS_SECRET_ACCESS_KEY_PATH")
```

Comment:
is it case insensitive for aws keys? on line 31 in previous file they are miniscule letters.

Reply:
Yes but these evns come from https://github.com/openshift/release/blob/master/ci-operator/config/openshift/lightspeed-service/openshift-lightspeed-service-main.yaml#L74-L77
These envs are read the the content of the aws-env is whatever is within those files and the keys are based on the names of the files instead of the envs.